### PR TITLE
Simple computer player

### PR DESCRIPTION
**Pull request by junglie85**
Added a simple computer player that picks random positions. Added ability to play human vs human, human vs computer or computer vs computer to the text client. Also added ability to quit to the text client. Refactored into modules to better encapsulate different responsibilities of the game and text client.

### CODE REVIEW

File: lib/game/board.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    def position_valid?(position)
+      is_integer?(position) && available_positions.include?(position.to_i)
+    end
+
+    def validation_reason(position)
+      !is_integer?(position) || is_outside_board_range?(position) ? :invalid_position : :position_taken
+    end
+
+    def is_integer?(position)
```
Comment (ssunday) [inline lib/game/board.rb:36]: In ruby, usually we don't have the `is_` prefix as the question mark works

File: lib/game/strategy.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+require 'game/easy_strategy'
+require 'game/human_strategy'
+
+module Game
+  class Strategy
+    def self.create(name)
+      case name
+      when :human
+        HumanStrategy.new()
```
Comment (ssunday) [inline lib/game/strategy.rb:9]: Nice

File: spec/game/board_spec.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      it "does not have any errors when token placed in valid position" do
+        board = @board.place_token(0, Board::PLAYER_ONE)
+        expect(board.has_error?).to be false
+        expect(board.error).to be nil
+      end
+
+      it "can place a valid integer passed as a string" do
+        board = @board.place_token("0", Board::PLAYER_ONE)
+        expect(board.has_error?).to be false
+        expect(board.error).to be nil
```
Comment (ssunday) [inline spec/game/board_spec.rb:57]: `be_nil`

File: lib/game/strategy.rb (hunk 2 above)
Comment (junglie85) [reply lib/game/strategy.rb:9]: Thanks

File: spec/game/board_spec.rb (hunk 3 above)
Comment (junglie85) [reply spec/game/board_spec.rb:57]: Cheers

File: lib/game/board.rb (hunk 1 above)
Comment (junglie85) [reply lib/game/board.rb:36]: Thanks, I'll bear that in mind.

**Pull request merged**